### PR TITLE
Fixes distribution issues so the C extension could be enabled by default.

### PR DESCRIPTION
**Pull request by cheqianh**
In order to enable C extension by default, this PR did below changes:

1. Changed back to string-number conversion (I comment the current math logic for reference and use the original string conversion way instead). Windows require a VS C++ build tool which raise some allocation errors. Using string conversion works well. 
2. Changed `git submodule` command to `git clone`.
3. Removed jsonconversion from [setup.py's install_requires](https://github.com/amzn/ion-python/blob/master/setup.py#L67), which is added into requirement.txt, This fails installation on windows with an error msg:`Package jsonconversion requires a different Python 3.7.8 not in <= 2.7`

I opened [an issue](https://github.com/amzn/ion-python/issues/186) for above three changes, not urgent but would be good to do more investigation in the future.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -64,7 +64,6 @@ def run_setup():
 
         install_requires=[
             'six',
-            'jsonconversion'
```
Comment (tgregg) [inline setup.py:67]: What are the implications of this? Does it break something else? Why was it added in the first place?

**Review by tgregg — COMMENTED** [review]
Can you go into more detail about the errors you're seeing when you use the number conversion logic that doesn't require intermediate strings? I'd feel much more comfortable temporarily disabling this for Windows only, if necessary, than removing it entirely. Even better would be to determine the source of the errors and swap logic based on OS in as granular a way as possible.

File: setup.py (hunk 1 above)
Comment (cheqianh) [reply setup.py:67]: Discussed offline, I'll create a new commit and comment detail there.

**Comment by cheqianh** [comment]
1. Removed install_requires, everything works well.
2. Switch to string conversion, commit ed54575 includes conversion logic. I also updated [issue 186](https://github.com/amzn/ion-python/issues/186)

**Review by tgregg — APPROVED** [review]

**Pull request merged**